src/main/scala/it/pagopa/interop/purposeprocess/service/impl/TenantManagementImpl.scala

```diff
@@ -2,33 +2,30 @@ package it.pagopa.interop.purposeprocess.service.impl
 
 import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
 import it.pagopa.interop.purposeprocess.service.TenantManagementService
-import it.pagopa.interop.tenantmanagement.client.invoker.ApiInvoker
+import it.pagopa.interop.tenantmanagement.client.invoker.{ApiInvoker => TenantManagementInvoker, ApiError, BearerToken}
 import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
-import it.pagopa.interop.tenantmanagement.client.api.{TenantApi, EnumsSerializers}
-import it.pagopa.interop.tenantmanagement.client.invoker.BearerToken
+import it.pagopa.interop.tenantmanagement.client.api.TenantApi
 import it.pagopa.interop.tenantmanagement.client.model.Tenant
+import it.pagopa.interop.purposeprocess.error.PurposeProcessErrors.TenantNotFound
+import it.pagopa.interop.commons.utils.withHeaders
 
 import java.util.UUID
-import scala.concurrent.{Future, ExecutionContextExecutor}
-import it.pagopa.interop.tenantmanagement.client.invoker.ApiRequest
-import akka.actor.typed.ActorSystem
-import it.pagopa.interop.commons.utils.withHeaders
+import scala.concurrent.{Future, ExecutionContext}
 
-class TenantManagementServiceImpl(tenantManagementUrl: String, blockingEc: ExecutionContextExecutor)(implicit
-  system: ActorSystem[_]
+final case class TenantManagementServiceImpl(invoker: TenantManagementInvoker, api: TenantApi)(implicit
```

Comment:
Our code base has a mixed approach to the class definition, in particular:
- class Foo
- final class Foo
- final case class Foo
I suggest using `final class Foo` to service implementation. 
I would keep only the `final case class Foo` for the type definition.
@galales what do you think?

Reply:
it works for me 👍